### ISSUE TITLE
Any advice for training on other datasets?

### ISSUE THREAD

**Issue by ys-zong** [spyrosavl/Fairness-by-Learning-Orthogonal-Disentangled-Representations#19]
Hi authors,

Thanks for the nice work! The code works well on the given datasets, but when I tried to apply the method to other datasets, the accuracy greatly fluctuates and the model can not converge (I'm using the Cifar model). The performance is strange, for a binary classification task, in some epochs, it predicts almost all samples to 0, and in the next epoch, it predicts all samples to 1. I tried commenting other losses and only keeping the target classification loss to debug, but it does not converge either. I am wondering do you have any suggestions for it, like how to set the hyperparameters, etc? Thank you!

**Comment by spyrosavl** [comment]
Hi @ys-zong, I am not sure why this is happening with your dataset. The parameters we used can be found in the config files in the root folder. For example parameters for CIFAR-10 can be found [here](https://github.com/spyrosavl/Fairness-by-Learning-Orthogonal-Disentangled-Representations/blob/master/config-cifar10.json).

**Comment by ys-zong** [comment]
Thanks for your quick reply! Another thing I'd like to make sure of is can I directly use the `y_zt` [here](https://github.com/spyrosavl/Fairness-by-Learning-Orthogonal-Disentangled-Representations/blob/4d743d098e0fa247b81b9546f08e04554681c0fc/model/model.py#L301) with sigmoid/softmax to evaluate the prediction accuracy, or have do I have to fit an MLP using the representation for the target task?

**Comment by ys-zong** [comment]
Found the reason: the value of the weight decay is too large for my dataset. It works well after adjusting it
